### PR TITLE
Encaps Overhead & Multiple/Redundant Encryptions

### PR DESCRIPTION
**Pull request by boucadair**
A variety of encaps are already in place. Solutions that do not require an extra encap/overhead will be superior.

This is important to keep in mind, IMO.

### CODE REVIEW

**Review by SpencerDawkins — COMMENTED** [review]
(Yeah, that's about what I thought the complete protocol stack would look like :-)

ISTM that this text is heading towards 3GPP ATSSS design considerations, but we might not need much detail about that in order to motivate IETF work on IETF-controlled protocols. So, I'd suggest a rethink about what we must say, and what we don't need to say in this document. 

Thoughts?

**Comment by boucadair** [comment]
This is a fair comment, Spencer.  We need to agree what details to include and which one to leave out. We can even leave all the text out if we decide so :-)

My intent was to anticipate comments saying "all what you need is to setup tunnels via each of your access networks". With this text, we explain that adding another tunneling is not that optimal.

**Comment by mpiraux** [comment]
I also think this level of details is not needed at this stage. We can reconsiderate at later stages of the document.

**Comment by obonaventure** [comment]
I'm also not convinced that we should provide this level of details in an IETF document. We should leave it sufficiently simple and listing all the encapsulation will appear complex to some IETFers

**Comment by boucadair** [comment]
Updated this PR to reflect the outcome of our confcall: record that an overhead will be experienced + multiple encryptions.

**Review by mpiraux — APPROVED** [review]

File: draft-quic-atsss-reqs.mkd [hunk 1]
```diff
@@ -612,6 +612,18 @@ Another use case in scope for the MASQUE group is IP proxying which can be used
...
+
+In 3GPP architectures, a variety of encapsulations are already used to convey user data. The use of QUIC as a tunneling method for ATSSS purposes will thus add an extra overhead. This overhead is even exacerbated when the user data is forwarded over a non-3GPP network access given than an IPsec tunnel is used to graft the UE to the 5G Core Network. For example, the overhead without taking into account the QUIC tunnel is at least 142 octets for an IPv6 packet forwarded over a non-3GPP network access. As a reminder, ATSSSv1 does not add any overhead. 
+
+More investigation is required to assess whether the ATSSSv2 overhead is an issue. Solutions to optimize that overhead might be considered, if needed. 
+
+## Multiple/Redundant Encryptions
+
+QUIC data that matches an ATSSS rule will be subject to two encryptions (i.e., the one required with the remote peer and the one used to steer data via an ATSSS Proxy) when forwarded via a 3GPP network access. Three encryptions will be imposed on the same data when forwarded over a non-3GPP network access (see {{sec-overhead}}). Some of these encryptions are redundant (e.g., ATSSS encryption vs. IP Encapsulating Security Payload (ESP)).
```
Comment (mpiraux) [inline draft-quic-atsss-reqs.mkd]:
```suggestion
QUIC data that matches an ATSSS rule will be subject to two encryptions (i.e., the one required with the remote peer and the one used to steer data via an ATSSS UPF) when forwarded via a 3GPP network access. Three encryptions will be imposed on the same data when forwarded over a non-3GPP network access (see {{sec-overhead}}). Some of these encryptions are redundant (e.g., ATSSS encryption vs. IP Encapsulating Security Payload (ESP)).
```

**Pull request merged**